core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java

```diff
@@ -223,4 +229,35 @@ public void testInvalidIndexName() {
                     e.getMessage().contains("Invalid index name [..], must not be \'.\' or '..'"), equalTo(true));
         }
     }
+
+    /**
+     * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
```

Comment:
Two unit tests, one for replication operation the test it listens to the request and one for TransportReplicationAction that tests it resolve request based on index meta data is way better then an integration test that does way more in terms of setup and tear downs

On 29 jul. 2016 11:05 PM +0200, Ali Beyad <notifications@github.com>, wrote:
>
> In core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java (https://github.com/elastic/elasticsearch/pull/19454#discussion_r72860336):
>
> > @@ -223,4 +229,35 @@ public void testInvalidIndexName() { > e.getMessage().contains("Invalid index name [..], must not be \'.\' or '..'"), equalTo(true)); > } > } > + > + /** > + * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}. > + */ > + public void testChangeWaitForActiveShardsSetting() throws Exception {
>
> I suppose it doesn't have to be, but I wanted to test not just the setting changing but that the replication operation uses the new setting... so this was the most convenient way to achieve that. I have a similar test for index creation in CreateIndexIT
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (https://github.com/elastic/elasticsearch/pull/19454/files/bcfd58f1bbac5e0dc7a65fa9365821ef0b39c052#r72860336), or mute the thread (https://github.com/notifications/unsubscribe-auth/AA9bJ-M8mebc58pUmq-E8eEjqV5wlHtKks5qamr-gaJpZM4JNuiL).
>
>
>